test/signalrclient-testhost/Connections/HubConnection.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNet.SignalR;
+
+namespace SelfHost
+{
+    public class HubConnection : Hub
+    {
+        public IEnumerable<int> ForceReconnect()
+        {
+            yield return 1;
+            // throwing here will close the websocket which should trigger reconnect
+            throw new Exception();
+        }
+
+        public string DisplayMessage(string message)
+        {
+            Clients.Caller.displayMessage("Send: " + message);
```

Comment:
You are doing two things in this hub which result in two separate actions in test. I would separate returning the value and invoking the client function and would create two separate tests for testing each scenario. Otherwise if a test fails it will be harder to figure out why it failed.